### PR TITLE
commit corrections search organization

### PR DESCRIPTION
**Pull request by JayGibran**
Corrections in organization search. 
It was added new parameters to method and query.

### CODE REVIEW

File: src/main/java/org/c4sg/dao/OrganizationDAO.java [hunk 1]
```diff
@@ -17,12 +17,45 @@
...
+            "OR LOWER(o.description) LIKE LOWER(CONCAT('%', :keyword, '%'))" + 
+            "OR LOWER(o.state) LIKE LOWER(CONCAT('%', :keyword, '%'))"+
+            "OR LOWER(o.country) LIKE LOWER(CONCAT('%', :keyword, '%')))" + 
+            "AND (o.country in (:country))" +
+            "AND (:status is null OR o.status = :status) "+
+            "AND (:category is null OR o.category = :category) " +
+            "AND p.id is null "+
+            " ORDER BY o.name ASC";
+    
+    String FIND_BY_KEYWORD_COUNTRIES_OPORTUNITES = "SELECT DISTINCT o FROM Project p INNER JOIN p.organization o " +
```
Comment (felipealvesgnu) [inline src/main/java/org/c4sg/dao/OrganizationDAO.java:50]: Delete those FIND_BY..... and replace by Query methods as mentioned here: https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#jpa.query-methods.query-creation

File: src/main/java/org/c4sg/service/impl/OrganizationServiceImpl.java [hunk 2]
```diff
@@ -109,18 +110,11 @@ public OrganizationDTO updateOrganization(int id, OrganizationDTO organizationDT
...
-    		List<ProjectDTO> projects=projectService.findByOrganization(id);
-    		for (ProjectDTO project:projects){
-    			projectService.deleteProject(project.getId());
-    		}
-    		organizationDAO.deleteUserOrganizations(id);
     		//TODO: Local or Timezone?
     		//TODO: Format date
     		//organization.setDeleteTime(LocalDateTime.now().toString());
     		//organization.setDeleteBy(user.getUsername());
+    		organizationDAO.save(organization);
```
Comment (felipealvesgnu) [inline src/main/java/org/c4sg/service/impl/OrganizationServiceImpl.java:117]: Why are you calling `organizationDAO.save(organization)` method if the method is called `deleteOrganization(int i)` ?

**Review by felipealvesgnu — CHANGES_REQUESTED** [review]

File: src/main/java/org/c4sg/dao/OrganizationDAO.java (hunk 1 above)
Comment (JayGibran) [reply src/main/java/org/c4sg/dao/OrganizationDAO.java:50]: Hi Felipe, do you talk about remove static strings and use directly in the query ?

Comment (felipealvesgnu) [reply src/main/java/org/c4sg/dao/OrganizationDAO.java:50]: Hi..No, I said to use the query creation mechanism as mentioned in the link I attached to you. You can follow the examples that we have in the project as well.

**Review by joshisayali — COMMENTED** [review]
queries FIND_BY_KEYWORD and FIND_BY_KEYWORD_COUNTRIES are returning wrong results. As per my understanding, it should return all the organizations with or without opportunities. Please get it clarified. Also please get clarified what should be the query result when open = true, open = false and value for open is not provided.

**Comment by felipealvesgnu** [comment]
@JayGibran are you able to fix those codes ?

**Pull request closed without merging**